src/gui/plot/qgsplotcanvas.cpp

```diff
@@ -0,0 +1,323 @@
+}
+
+void QgsPlotCanvas::mousePressEvent( QMouseEvent *event )
+{
+  if ( mTool )
+  {
+    std::unique_ptr<QgsPlotMouseEvent> me( new QgsPlotMouseEvent( this, event ) );
+    mTool->plotPressEvent( me.get() );
+    event->setAccepted( me->isAccepted() );
+  }
+
+  if ( !mTool || !event->isAccepted() )
+  {
+    if ( event->button() == Qt::MiddleButton )
+    {
+      // Pan layout with middle mouse button
+      setTool( mMidMouseButtonPanTool );
+      event->accept();
+    }
+    else if ( event->button() == Qt::RightButton && mTool->flags() & Qgis::PlotToolFlag::ShowContextMenu )
+    {
+      std::unique_ptr<QgsPlotMouseEvent> me( new QgsPlotMouseEvent( this, event ) );
+      showContextMenu( me.get() );
+      return;
+    }
+    else
+    {
+      QGraphicsView::mousePressEvent( event );
+    }
+  }
+}
+
+void QgsPlotCanvas::mouseReleaseEvent( QMouseEvent *event )
+{
+  if ( mTool )
+  {
+    std::unique_ptr<QgsPlotMouseEvent> me( new QgsPlotMouseEvent( this, event ) );
+    mTool->plotReleaseEvent( me.get() );
+    event->setAccepted( me->isAccepted() );
+  }
+
+  if ( !mTool || !event->isAccepted() )
+    QGraphicsView::mouseReleaseEvent( event );
+}
+
+void QgsPlotCanvas::resizeEvent( QResizeEvent *e )
+{
+  QGraphicsView::resizeEvent( e );
+}
+
+void QgsPlotCanvas::wheelEvent( QWheelEvent *event )
+{
+  if ( mTool )
+  {
+    mTool->wheelEvent( event );
+  }
+
+  if ( !mTool || !event->isAccepted() )
+  {
+    event->accept();
+    wheelZoom( event );
+  }
+}
+
+void QgsPlotCanvas::mouseMoveEvent( QMouseEvent *event )
+{
+  if ( mTool )
+  {
+    std::unique_ptr<QgsPlotMouseEvent> me( new QgsPlotMouseEvent( this, event ) );
+    mTool->plotMoveEvent( me.get() );
+    event->setAccepted( me->isAccepted() );
+  }
+
+  if ( !mTool || !event->isAccepted() )
+    QGraphicsView::mouseMoveEvent( event );
+}
+
+void QgsPlotCanvas::setTool( QgsPlotTool *tool )
+{
+  if ( mTool )
+  {
+    mTool->deactivate();
+  }
+
+  if ( tool )
+  {
+    // activate new tool before setting it - gives tools a chance
+    // to respond to whatever the current tool is
+    tool->activate();
+  }
+
+  mTool = tool;
+  emit toolChanged( mTool );
+}
+
+void QgsPlotCanvas::unsetTool( QgsPlotTool *tool )
+{
+  if ( mTool && mTool == tool )
+  {
+    mTool->deactivate();
+    emit toolChanged( nullptr );
+    setCursor( Qt::ArrowCursor );
+  }
+}
+
+QgsPlotTool *QgsPlotCanvas::tool()
+{
+  return mTool;
+}
+
+QgsCoordinateReferenceSystem QgsPlotCanvas::crs() const
+{
+  return QgsCoordinateReferenceSystem();
+}
+
+QgsPoint QgsPlotCanvas::toMapCoordinates( const QgsPointXY & ) const
+{
+  return QgsPoint();
+}
+
+QgsPointXY QgsPlotCanvas::toCanvasCoordinates( const QgsPoint & ) const
+{
+  return QgsPointXY();
+}
+
+void QgsPlotCanvas::panContentsBy( double, double )
+{
+
+}
+
+void QgsPlotCanvas::centerPlotOn( double, double )
+{
+
+}
+
+void QgsPlotCanvas::scalePlot( double )
+{
+
+}
+
+void QgsPlotCanvas::zoomToRect( const QRectF )
+{
+
+}
```

Comment:
are these intentionally left empty?

Reply:
Yes -- the base plot class doesn't support these and they must be implemented in the subclasses if applicable